### ISSUE TITLE
inquiry about StructuredAttention.py

### ISSUE THREAD

**Issue by FayeXXX** [vidhishanair/structured-text-representations#2]
hi，thanks for sharing that excellent code. I notice that there is a "LL_inv = tf.matrix_inverse" in the origin code for structured attention, and u have changed it to line 75 to 79 in StructuredAttention.py. Could u plz tell me why u change it and how it works? Thank u a lot.

**Comment by vidhishanair** [comment]
Hi,
So the equivalent of matrix_inverse in Pytorch is torch.inverse(). This code was released when batched torch.inverse was only available in the nightly version. As of today, torch.inverse is available in Pytorch 1.6, so I will update the code to use this by default. For now you can use it by setting pytorch_version as 'nightly'. Before the batched code was made available, the only option was to perform inverse for each example and then stack it, which is what the else part does.

**Comment by FayeXXX** [comment]
> Hi,
> So the equivalent of matrix_inverse in Pytorch is torch.inverse(). This code was released when batched torch.inverse was only available in the nightly version. As of today, torch.inverse is available in Pytorch 1.6, so I will update the code to use this by default. For now you can use it by setting pytorch_version as 'nightly'. Before the batched code was made available, the only option was to perform inverse for each example and then stack it, which is what the else part does.

Thank u for ur quik reply. And I've tried to use torch.inverse, but actually it may be a mathematical problem. The determinant of L_ij_bar is zero, so it cannot be inversed. So how to fix it?

**Comment by vidhishanair** [comment]
Sorry for the delay! There isn't a fix to that. In this case, this happens only with random seeds. So re-training with different seed helps!
